sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py

```diff
@@ -0,0 +1,433 @@
+        :type partition_key: str
+        :param row_key: The row key of the entity.
+        :type row_key: str
+        :param etag: Etag of the entity
+        :type etag: str
+        :param match_condition: MatchCondition
+        :type match_condition: ~azure.core.MatchConditions
+        :return: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=etag, match_condition=match_condition),
+                                                    etag_param='etag', match_param='match_condition')
+
+        if table_entity_properties:
+            partition_key = table_entity_properties['PartitionKey']
+            row_key = table_entity_properties['RowKey']
+            table_entity_properties = _add_entity_properties(table_entity_properties)
+
+        if mode is UpdateMode.REPLACE:
+            await self._client.table.update_entity(
+                table=self.table_name,
+                partition_key=partition_key,
+                row_key=row_key,
+                table_entity_properties=table_entity_properties,
+                if_match=if_match or if_not_match or "*",
+                **kwargs)
+        if mode is UpdateMode.MERGE:
+            await self._client.table.merge_entity(table=self.table_name, partition_key=partition_key,
+                                                  row_key=row_key, if_match=if_match or if_not_match or "*",
+                                                  table_entity_properties=table_entity_properties, **kwargs)
+
+    @distributed_trace
+    def query_entities(
+            self,
+            results_per_page=None,
+            select=None,
+            filter=None,  # pylint: disable=W0622
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> ItemPaged[TableEntity]
+        """Queries entities in a table.
+        :param results_per_page: Number of entities per page in return ItemPaged
+        :type results_per_page: int
+        :param select: Specify desired properties of an entity to return certain entities
+        :type select: str
+        :param filter: Specify a filter to return certain entities
+        :type filter: str
+        :return: Query of table entities
+        :rtype: ItemPaged[TableEntity]
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        query_options = QueryOptions(top=results_per_page, select=select, filter=filter)
+
+        command = functools.partial(
+            self._client.table.query_entities,
+            **kwargs)
+        return AsyncItemPaged(
+            command, results_per_page=query_options, table=self.table_name,
+            page_iterator_class=TableEntityPropertiesPaged
+        )
+
+    @distributed_trace_async
+    async def get_entity(
+            self,
+            partition_key,  # type: str
+            row_key,  # type: str
+            results_per_page=None,
```

Comment:
I think we removed query_options from get_entity as it only gets 1 entity .. so you can remove filter, results_per_page, select